Semantic Release

andnp opened the pull request:
Semantic release will controll the semver of each release based on the
tags given at the beginning of each commit (which is now required by
the commit linter). This will allow for automated releases in travis
and will additionally remove the guesswork for version numbers.

Commit:
chore: use husky for git hooks, add commit linting

Commit:
build: publish using semantic-relase

Semantic release will controll the semver of each release based on the
tags given at the beginning of each commit (which is now required by
the commit linter). This will allow for automated releases in travis
and will additionally remove the guesswork for version numbers.

Retsam commented:
I'm not sure if it's intentional or not, but there's lots of commits in here that seem unrelated to the Semantic Release changes.  

Husky looks neat, though I think I prefer the syntax from the 0.14 (beta) version of `husky` which puts the hooks [in their own section of package.json](https://github.com/typicode/husky#upgrading-from-014), rather than mixing them into the scripts section.

andnp commented:
Oops, I rebased the parent branch and deleted several of the commits. That must have upset this branch quite a bit.

That's nice about having its own section. I had missed that!

Commit:
commit: move git hooks to husky property in package
